crates/bevy_color/src/color_gradient.rs

```diff
@@ -0,0 +1,104 @@
+use crate::Mix;
+use bevy_math::curve::{cores::EvenCoreError, Curve, Interval, SampleCurve};
+
+/// A curve whose samples are defined by a collection of colors.
+#[derive(Clone, Debug)]
+pub struct ColorCurve<T: Mix + Clone, I> {
+    curve: SampleCurve<T, I>,
+}
```

Comment:
```suggestion
pub struct ColorCurve<T: Mix + Clone> {
    curve: SampleCurve<T, fn(&T, &T, f32) -> T>,
}
```

We can replace the generic parameter `I` with a concrete function pointer type instead (`fn` instead of `Fn`). This means we can't use a function which captures variables from the scope, but we already want to use a standard function (`Mix::mix`) so we haven't lost any utility doing this.